[Upgrade Assistant] Check for ML upgrade mode before enabling flyout actions

sabarasaba opened the pull request:
Fixes https://github.com/elastic/kibana/issues/112100

### Summary

This PR adds checks if the ml_upgrade_mode is enabled, and if so we disable the flyout actions and show an error callout informing the user that is not possible to do any actions while its enabled.

**How to test**
1. Start an elasticsearch instance using a snapshot from 6.8. Alison's [snapshot](https://github.com/elastic/kibana/pull/107053) worked just fine for me
2. Start kibana with `yarn start`
3. Run `POST _ml/set_upgrade_mode?enabled=true` in console
4. Navigate to UA, open `ES deprecation issues` and open flyout for a machine learning snapshot.

**Screenshots**

<details>
<summary>Overview of changes</summary>

// upgrade_mode=true
![Screenshot 2021-09-20 at 16 47 41](https://user-images.githubusercontent.com/1191206/134022902-ccf50ffb-c767-4d01-9013-022f643f65cb.png)

// upgrade_mode=false
![Screenshot 2021-09-20 at 15 28 16](https://user-images.githubusercontent.com/1191206/134012124-27ef6398-074b-450b-a2f6-f82cfcc925c1.png)
</details>

kibanamachine commented:
## :green_heart: Build Succeeded
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/154242/)
* Commit: e94548823a4fa3cc671b62e7122a9e56a5e0ab33
* [Storybooks Preview](https://ci-artifacts.kibana.dev/storybooks/pr-112555/e94548823a4fa3cc671b62e7122a9e56a5e0ab33)

### Metrics [[docs](https://www.elastic.co/guide/en/kibana/master/ci-metrics.html)]
#### Async chunks
> Total size of all lazy-loaded chunks that will be downloaded as the user navigates the app


| id | [before](https://github.com/elastic/kibana/commit/c16480e849389cc064ecae012dc925b0eae348b5) | [after](https://github.com/elastic/kibana/commit/e94548823a4fa3cc671b62e7122a9e56a5e0ab33) | diff |
| --- | --- | --- | --- |
| `upgradeAssistant` | 210.9KB | 212.7KB | +1.8KB |

#### Page load bundle
> Size of the bundles that are downloaded on every page load. Target size is below 100kb


| id | [before](https://github.com/elastic/kibana/commit/c16480e849389cc064ecae012dc925b0eae348b5) | [after](https://github.com/elastic/kibana/commit/e94548823a4fa3cc671b62e7122a9e56a5e0ab33) | diff |
| --- | --- | --- | --- |
| `core` | 407.2KB | 407.3KB | +61.0B |
| `upgradeAssistant` | 25.0KB | 25.1KB | +135.0B |
| total |  |  | +196.0B |

To update your PR or re-run it, just comment with:
`@elasticmachine merge upstream`

cc @sabarasaba

<!--PIPELINE
{"builds":[{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/154242/","number":"154242","commit":"e94548823a4fa3cc671b62e7122a9e56a5e0ab33"}]}
PIPELINE-->

kibanamachine commented:
## :green_heart: Build Succeeded
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/154267/)
* Commit: 55422a8e0611e875279b53020d220b2ec0d4b825
* [Storybooks Preview](https://ci-artifacts.kibana.dev/storybooks/pr-112555/55422a8e0611e875279b53020d220b2ec0d4b825)

### Metrics [[docs](https://www.elastic.co/guide/en/kibana/master/ci-metrics.html)]
#### Async chunks
> Total size of all lazy-loaded chunks that will be downloaded as the user navigates the app


| id | [before](https://github.com/elastic/kibana/commit/c16480e849389cc064ecae012dc925b0eae348b5) | [after](https://github.com/elastic/kibana/commit/55422a8e0611e875279b53020d220b2ec0d4b825) | diff |
| --- | --- | --- | --- |
| `upgradeAssistant` | 210.9KB | 212.7KB | +1.8KB |

#### Page load bundle
> Size of the bundles that are downloaded on every page load. Target size is below 100kb


| id | [before](https://github.com/elastic/kibana/commit/c16480e849389cc064ecae012dc925b0eae348b5) | [after](https://github.com/elastic/kibana/commit/55422a8e0611e875279b53020d220b2ec0d4b825) | diff |
| --- | --- | --- | --- |
| `core` | 407.2KB | 407.3KB | +61.0B |
| `upgradeAssistant` | 25.0KB | 25.1KB | +135.0B |
| total |  |  | +196.0B |

### History
* :green_heart: [Build #154242](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/154242/) succeeded e94548823a4fa3cc671b62e7122a9e56a5e0ab33

To update your PR or re-run it, just comment with:
`@elasticmachine merge upstream`

cc @sabarasaba

<!--PIPELINE
{"builds":[{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/154242/","number":"154242","commit":"e94548823a4fa3cc671b62e7122a9e56a5e0ab33"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/154267/","number":"154267","commit":"55422a8e0611e875279b53020d220b2ec0d4b825"}]}
PIPELINE-->

kibanamachine commented:
## :green_heart: Build Succeeded
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/154315/)
* Commit: 45e26b84ebb397c0d0d237555e2e03729176ac9e
* [Storybooks Preview](https://ci-artifacts.kibana.dev/storybooks/pr-112555/45e26b84ebb397c0d0d237555e2e03729176ac9e)

### Metrics [[docs](https://www.elastic.co/guide/en/kibana/master/ci-metrics.html)]
#### Async chunks
> Total size of all lazy-loaded chunks that will be downloaded as the user navigates the app


| id | [before](https://github.com/elastic/kibana/commit/c16480e849389cc064ecae012dc925b0eae348b5) | [after](https://github.com/elastic/kibana/commit/45e26b84ebb397c0d0d237555e2e03729176ac9e) | diff |
| --- | --- | --- | --- |
| `upgradeAssistant` | 210.9KB | 212.9KB | +2.0KB |

#### Page load bundle
> Size of the bundles that are downloaded on every page load. Target size is below 100kb


| id | [before](https://github.com/elastic/kibana/commit/c16480e849389cc064ecae012dc925b0eae348b5) | [after](https://github.com/elastic/kibana/commit/45e26b84ebb397c0d0d237555e2e03729176ac9e) | diff |
| --- | --- | --- | --- |
| `core` | 407.2KB | 407.3KB | +61.0B |
| `upgradeAssistant` | 25.0KB | 25.1KB | +135.0B |
| total |  |  | +196.0B |

### History
* :green_heart: [Build #154267](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/154267/) succeeded 55422a8e0611e875279b53020d220b2ec0d4b825
* :green_heart: [Build #154242](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/154242/) succeeded e94548823a4fa3cc671b62e7122a9e56a5e0ab33

To update your PR or re-run it, just comment with:
`@elasticmachine merge upstream`

cc @sabarasaba

<!--PIPELINE
{"builds":[{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/154242/","number":"154242","commit":"e94548823a4fa3cc671b62e7122a9e56a5e0ab33"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/154267/","number":"154267","commit":"55422a8e0611e875279b53020d220b2ec0d4b825"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/154315/","number":"154315","commit":"45e26b84ebb397c0d0d237555e2e03729176ac9e"}]}
PIPELINE-->

elasticmachine commented:
Pinging @elastic/kibana-stack-management (Team:Stack Management)

cjcenizal commented on x-pack/plugins/upgrade_assistant/__jest__/client_integration/es_deprecations/deprecations_list.test.ts:
```diff
@@ -63,16 +63,16 @@ describe('ES deprecations table', () => {
     const reindexDeprecation = esDeprecationsMockResponse.deprecations[3];
 
     // Since upgradeStatusMockResponse includes ML and reindex actions (which require fetching status), there will be 3 requests made
```
This comment seems out of date since it refers to 3 requests, but the code now expects 4 more requests. Can we update it?

cjcenizal commented on x-pack/plugins/upgrade_assistant/__jest__/client_integration/es_deprecations/deprecations_list.test.ts:
```diff
@@ -63,16 +63,16 @@ describe('ES deprecations table', () => {
       }`
     );
-    expect(server.requests[server.requests.length - 1].url).toBe(
+    expect(server.requests[server.requests.length - 2].url).toBe(
```
Do we also want to verify what the latest request is, i.e. `server.requests.length - 1`?

sabarasaba replied:
Nice catch!

cjcenizal commented on x-pack/plugins/upgrade_assistant/__jest__/client_integration/es_deprecations/ml_snapshots_deprecation_flyout.test.ts:
```diff
@@ -131,6 +132,27 @@ describe('Machine learning deprecation flyout', () => {
       expect(find('mlSnapshotDetails.upgradeSnapshotButton').text()).toEqual('Retry upgrade');
     });
+
+    it('Doesnt allow to take actions if ml_upgrade_mode is enabled', async () => {
```
Nit: `Doesnt allow to take` might be more clearly stated as `Disables`

cjcenizal commented on x-pack/plugins/upgrade_assistant/public/application/lib/api.ts:
```diff
@@ -180,6 +180,15 @@ export class ApiService {
     });
   }
+
+  public useLoadMLUpgradeMode() {
```
Nit: Can we retain the camelcasing pattern by naming this `useLoadMlUpgradeMode`?

cjcenizal commented on x-pack/plugins/upgrade_assistant/server/routes/ml_snapshots.test.ts:
```diff
@@ -175,6 +175,28 @@ describe('ML snapshots APIs', () => {
   });
 
+  describe('GET /api/upgrade_assistant/ml_upgrade_mode', () => {
+    it('Knows how to retrieve ml upgrade mode', async () => {
```
Perhaps clearer to state "Retrieves ML upgrade mode"?

cjcenizal commented on x-pack/plugins/upgrade_assistant/public/application/components/es_deprecations/deprecation_types/ml_snapshots/flyout.tsx:
```diff
@@ -97,6 +99,28 @@ const i18nTexts = {
+          <EuiLink href={docsLink} target="_blank" data-test-subj="setUpgradeModeDocsLink">
+            <FormattedMessage
+              id="xpack.upgradeAssistant.esDeprecations.mlSnapshots.upgradeModeEnabledDocsLink"
+              defaultMessage="Documentation"
```
I think this should be "Learn more".

cjcenizal reviewed (approved):
Code LGTM. I had a few nitpicks, you know me. :)

I couldn't test locally though, I kept getting security errors:

```
[error][savedobjects-service] [.kibana] Action failed with 'security_exception: [security_exception] Reason: current license is non-compliant for [security]'
```

kibanamachine commented:
## :green_heart: Build Succeeded
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/154847/)
* Commit: 5b76125e082c658c9364bac2392a8d082606a9f3
* [Storybooks Preview](https://ci-artifacts.kibana.dev/storybooks/pr-112555/5b76125e082c658c9364bac2392a8d082606a9f3)

### Metrics [[docs](https://www.elastic.co/guide/en/kibana/master/ci-metrics.html)]
#### Async chunks
> Total size of all lazy-loaded chunks that will be downloaded as the user navigates the app


| id | [before](https://github.com/elastic/kibana/commit/c16480e849389cc064ecae012dc925b0eae348b5) | [after](https://github.com/elastic/kibana/commit/5b76125e082c658c9364bac2392a8d082606a9f3) | diff |
| --- | --- | --- | --- |
| `upgradeAssistant` | 210.9KB | 212.9KB | +2.0KB |

#### Page load bundle
> Size of the bundles that are downloaded on every page load. Target size is below 100kb


| id | [before](https://github.com/elastic/kibana/commit/c16480e849389cc064ecae012dc925b0eae348b5) | [after](https://github.com/elastic/kibana/commit/5b76125e082c658c9364bac2392a8d082606a9f3) | diff |
| --- | --- | --- | --- |
| `core` | 407.2KB | 407.3KB | +61.0B |
| `upgradeAssistant` | 25.0KB | 25.1KB | +135.0B |
| total |  |  | +196.0B |

### History
* :green_heart: [Build #154315](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/154315/) succeeded 45e26b84ebb397c0d0d237555e2e03729176ac9e
* :green_heart: [Build #154267](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/154267/) succeeded 55422a8e0611e875279b53020d220b2ec0d4b825
* :green_heart: [Build #154242](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/154242/) succeeded e94548823a4fa3cc671b62e7122a9e56a5e0ab33

To update your PR or re-run it, just comment with:
`@elasticmachine merge upstream`

cc @sabarasaba

<!--PIPELINE
{"builds":[{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/154242/","number":"154242","commit":"e94548823a4fa3cc671b62e7122a9e56a5e0ab33"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/154267/","number":"154267","commit":"55422a8e0611e875279b53020d220b2ec0d4b825"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/154315/","number":"154315","commit":"45e26b84ebb397c0d0d237555e2e03729176ac9e"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/154847/","number":"154847","commit":"5b76125e082c658c9364bac2392a8d082606a9f3"}]}
PIPELINE-->

sabarasaba commented:
@cjcenizal Thanks for having a look! I've addressed your comments, and in regards of testing it you can also try with [the steps](https://github.com/elastic/kibana/tree/7.x/x-pack/plugins/upgrade_assistant#elasticsearch-deprecations) listed in the readme

alisonelizabeth commented on x-pack/plugins/upgrade_assistant/public/application/components/es_deprecations/deprecation_types/ml_snapshots/flyout.tsx:
```diff
@@ -97,6 +99,28 @@ const i18nTexts = {
+  upgradeModeEnabledErrorTitle: i18n.translate(
+    'xpack.upgradeAssistant.esDeprecations.mlSnapshots.upgradeModeEnabledErrorTitle',
+    {
+      defaultMessage: 'Machine learning upgrade mode is enabled',
```
```suggestion
      defaultMessage: 'Machine Learning upgrade mode is enabled',
```

alisonelizabeth commented on x-pack/plugins/upgrade_assistant/public/application/components/es_deprecations/deprecation_types/ml_snapshots/flyout.tsx:
```diff
@@ -97,6 +99,28 @@ const i18nTexts = {
+  upgradeModeEnabledErrorDescription: (docsLink: string) => (
+    <FormattedMessage
+      id="xpack.upgradeAssistant.esDeprecations.mlSnapshots.upgradeModeEnabledErrorDescription"
+      defaultMessage="No actions can be taken on machine learning snapshots while upgrade mode is enabled. {docsLink}."
```
```suggestion
      defaultMessage="No actions can be taken on Machine Learning snapshots while upgrade mode is enabled. {docsLink}."
```

alisonelizabeth commented on x-pack/plugins/upgrade_assistant/public/application/components/es_deprecations/es_deprecations_table.tsx:
```diff
@@ -146,6 +156,13 @@ export const EsDeprecationsTable: React.FunctionComponent<Props> = ({
+    services: { api },
+  } = useAppContext();
+
+  const { data } = api.useLoadMlUpgradeMode();
```
What are your thoughts on fetching this within the flyout? If I'm not mistaken, this request will be made every time the ES deprecations page is loaded, even if there are no ML deprecations.

Also, is there anything we have to consider if the user has a `basic` license, or should this still work as expected? I also see there are [privileges](https://www.elastic.co/guide/en/elasticsearch/reference/current/ml-set-upgrade-mode.html#ml-set-upgrade-mode-prereqs) associated with the API.

sabarasaba replied:
I figured that if we fetch within the flyout we have two choices:

* We actually make the fetch when the flyout opens up, which will cause layout shifting due to callouts being rendered above the deprecation details
* Fetch the status in the `GET ml_snapshots/{jobId}/{snapshotId}` endpoint (my first implementation was based on this one), but that would generate an extra es api call per request. This one could work nicely and I don't mind changing the implementation to accommodate, but just wasn't sure how much we try to optimize for es api calls or if it's fast enough that it doesn't make much of a difference.

I don't think theres anything special we should do for `basic` license, I've done most of my testing using that one and it works just fine. 

Good catch in regards of the privileges! I don't think its a problem though, the only api we call is `GET _ml/info` for which the user should have the `monitor_ml` cluster privilege. At the moment if a user doesn't have it, the kibana api `GET upgrade_assistant/ml_upgrade_mode` will fail and the callout won't be rendered but actions will. Upon performing an action they will get an error accordingly. If we don't want it to fail, we could check for `monitor_ml` privileges before calling `GET _ml/info` and assume its disabled if we don't have privileges. WDYT?

alisonelizabeth replied:
Thanks for the explanation!

I understand the concerns. It just felt a little strange to me to make a request that may or may not be necessary. I think there are two conditions that would need to be met for it to be used: (1) user needs to have at least 1 ML deprecation and (2) user needs to open the ML deprecation flyout. 

That said, I’m happy to leave as-is for now and see if the current approach causes any issues.

alisonelizabeth replied:
As far as the second part -

> I don't think theres anything special we should do for basic license, I've done most of my testing using that one and it works just fine.

👍 Agreed - `GET _ml/info` seems to work fine. It looks like ES will return an error if a user for some reason has a basic license but then tries to upgrade/delete a snapshot (probably an edge case).

<img width="268" alt="Screen Shot 2021-09-23 at 10 06 28 AM" src="https://user-images.githubusercontent.com/5226211/134523528-2a7de145-df38-42f8-af09-15f81c25a8a0.png">

> Good catch in regards of the privileges! I don't think its a problem though, the only api we call is GET _ml/info for which the user should have the monitor_ml cluster privilege. At the moment if a user doesn't have it, the kibana api GET upgrade_assistant/ml_upgrade_mode will fail and the callout won't be rendered but actions will. Upon performing an action they will get an error accordingly. If we don't want it to fail, we could check for monitor_ml privileges before calling GET _ml/info and assume its disabled if we don't have privileges. WDYT?

👍 I think this is outside the scope of this PR, but I think it would be a nice-to-have to check privileges for the ML action, as I just noticed the reindexing action does show a callout if the user does not have correct privileges.

<img width="513" alt="Screen Shot 2021-09-23 at 10 06 12 AM" src="https://user-images.githubusercontent.com/5226211/134524220-ee4f9b42-0ed5-4c40-8711-70a16bd37b07.png">

alisonelizabeth reviewed (commented):
Just passing through 😄 . I had a question about where the ml upgrade mode request is made, and also if we need to consider anything around licensing/privileges.

kibanamachine commented:
## :green_heart: Build Succeeded
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/155212/)
* Commit: 865ff6ce99a7c7f769aca1398ad4b02ba83b00c1
* [Storybooks Preview](https://ci-artifacts.kibana.dev/storybooks/pr-112555/865ff6ce99a7c7f769aca1398ad4b02ba83b00c1)

### Metrics [[docs](https://www.elastic.co/guide/en/kibana/master/ci-metrics.html)]
#### Async chunks
> Total size of all lazy-loaded chunks that will be downloaded as the user navigates the app


| id | [before](https://github.com/elastic/kibana/commit/c16480e849389cc064ecae012dc925b0eae348b5) | [after](https://github.com/elastic/kibana/commit/865ff6ce99a7c7f769aca1398ad4b02ba83b00c1) | diff |
| --- | --- | --- | --- |
| `upgradeAssistant` | 210.9KB | 212.9KB | +2.0KB |

#### Page load bundle
> Size of the bundles that are downloaded on every page load. Target size is below 100kb


| id | [before](https://github.com/elastic/kibana/commit/c16480e849389cc064ecae012dc925b0eae348b5) | [after](https://github.com/elastic/kibana/commit/865ff6ce99a7c7f769aca1398ad4b02ba83b00c1) | diff |
| --- | --- | --- | --- |
| `core` | 407.2KB | 407.3KB | +61.0B |
| `upgradeAssistant` | 25.0KB | 25.1KB | +135.0B |
| total |  |  | +196.0B |

### History
* :green_heart: [Build #154847](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/154847/) succeeded 5b76125e082c658c9364bac2392a8d082606a9f3
* :green_heart: [Build #154315](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/154315/) succeeded 45e26b84ebb397c0d0d237555e2e03729176ac9e
* :green_heart: [Build #154267](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/154267/) succeeded 55422a8e0611e875279b53020d220b2ec0d4b825
* :green_heart: [Build #154242](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/154242/) succeeded e94548823a4fa3cc671b62e7122a9e56a5e0ab33

To update your PR or re-run it, just comment with:
`@elasticmachine merge upstream`

cc @sabarasaba

<!--PIPELINE
{"builds":[{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/154242/","number":"154242","commit":"e94548823a4fa3cc671b62e7122a9e56a5e0ab33"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/154267/","number":"154267","commit":"55422a8e0611e875279b53020d220b2ec0d4b825"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/154315/","number":"154315","commit":"45e26b84ebb397c0d0d237555e2e03729176ac9e"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/154847/","number":"154847","commit":"5b76125e082c658c9364bac2392a8d082606a9f3"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/155212/","number":"155212","commit":"865ff6ce99a7c7f769aca1398ad4b02ba83b00c1"}]}
PIPELINE-->

kibanamachine commented:
## :green_heart: Build Succeeded
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/155727/)
* Commit: 29d5f35f0bf586576cd9cfc702a1abb0d0d6a40c
* [Storybooks Preview](https://ci-artifacts.kibana.dev/storybooks/pr-112555/29d5f35f0bf586576cd9cfc702a1abb0d0d6a40c)

### Metrics [[docs](https://www.elastic.co/guide/en/kibana/master/ci-metrics.html)]
#### Async chunks
> Total size of all lazy-loaded chunks that will be downloaded as the user navigates the app


| id | [before](https://github.com/elastic/kibana/commit/9357d9ea9c43e730891d9c55b2d46f8103a5658a) | [after](https://github.com/elastic/kibana/commit/29d5f35f0bf586576cd9cfc702a1abb0d0d6a40c) | diff |
| --- | --- | --- | --- |
| `upgradeAssistant` | 212.3KB | 214.3KB | +2.0KB |

#### Page load bundle
> Size of the bundles that are downloaded on every page load. Target size is below 100kb


| id | [before](https://github.com/elastic/kibana/commit/9357d9ea9c43e730891d9c55b2d46f8103a5658a) | [after](https://github.com/elastic/kibana/commit/29d5f35f0bf586576cd9cfc702a1abb0d0d6a40c) | diff |
| --- | --- | --- | --- |
| `core` | 408.2KB | 408.3KB | +61.0B |
| `upgradeAssistant` | 28.3KB | 28.5KB | +135.0B |
| total |  |  | +196.0B |

### History
* :green_heart: [Build #155212](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/155212/) succeeded 865ff6ce99a7c7f769aca1398ad4b02ba83b00c1
* :green_heart: [Build #154847](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/154847/) succeeded 5b76125e082c658c9364bac2392a8d082606a9f3
* :green_heart: [Build #154315](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/154315/) succeeded 45e26b84ebb397c0d0d237555e2e03729176ac9e
* :green_heart: [Build #154267](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/154267/) succeeded 55422a8e0611e875279b53020d220b2ec0d4b825
* :green_heart: [Build #154242](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/154242/) succeeded e94548823a4fa3cc671b62e7122a9e56a5e0ab33

To update your PR or re-run it, just comment with:
`@elasticmachine merge upstream`

cc @sabarasaba

<!--PIPELINE
{"builds":[{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/154242/","number":"154242","commit":"e94548823a4fa3cc671b62e7122a9e56a5e0ab33"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/154267/","number":"154267","commit":"55422a8e0611e875279b53020d220b2ec0d4b825"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/154315/","number":"154315","commit":"45e26b84ebb397c0d0d237555e2e03729176ac9e"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/154847/","number":"154847","commit":"5b76125e082c658c9364bac2392a8d082606a9f3"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/155212/","number":"155212","commit":"865ff6ce99a7c7f769aca1398ad4b02ba83b00c1"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/155727/","number":"155727","commit":"29d5f35f0bf586576cd9cfc702a1abb0d0d6a40c"}]}
PIPELINE-->
